src/Univ2LpOracle.t.sol

```diff
@@ -425,4 +478,137 @@ contract UNIV2LPOracleTest is DSTest {
         ethDaiLPOracle.deny(address(this));  // Remove owner
         ethDaiLPOracle.diss(address(this));  // Attempt to remove caller from whitelist
     }
+
+    function test_eth_dai_price_change() public {
+        ethDaiLPOracle.poke();                            // Poke oracle
+        ethDaiLPOracle.kiss(address(this));
+        (bytes32 val, bool has) = ethDaiLPOracle.peep();  // Peep oracle price without caller being whitelisted
+        uint256 firstVal = uint256(val);
+
+        assertTrue(firstVal < 100 ether && firstVal > 50 ether); // 57327394135985707908 at time of test
+        assertTrue(has);
+
+        /*** Trade 10k DAI for ETH ***/
+        assertEq(IERC20(DAI).balanceOf(address(this)), 10_000 ether);
+        address[] memory path = new address[](2);
+        path[0] = DAI;
+        path[1] = WETH;
+        uint[] memory amounts = uniswap.swapExactTokensForTokens(
+            IERC20(DAI).balanceOf(address(this)), 0, path, address(this), now);
+        assertEq(amounts.length, 2);
+        assertEq(amounts[0], 10_000 ether);
+        assertEq(IERC20(DAI).balanceOf(address(this)), 0);
+        assertEq(IERC20(WETH).balanceOf(address(this)), amounts[1]);
+
+        uint256 diff = 
+            amounts[1] > 10_000 ether * WAD / ethPrice ? 
+            amounts[1] - 10_000 ether * WAD / ethPrice : 
+            10_000 ether * WAD / ethPrice - amounts[1];
+        assertTrue(diff * WAD / amounts[1] < 0.02 ether); // Less than 2% slippage
+
+        hevm.warp(now + 3600);
+
+        ethDaiLPOracle.poke();
+        (val, has) = ethDaiLPOracle.peep();
+        uint256 secondVal = uint256(val);
+        assertTrue(secondVal < 100 ether && secondVal > 50 ether); // 58502000047042694225 at time of test
+        assertTrue(has);
+
+        // Ensure the price changed after the oracle update
+        assertTrue(secondVal > firstVal);
+
+        /*** Trade $200m ETH for DAI ***/
+        uint256 ethBal = IERC20(WETH).balanceOf(address(this));
+        path = new address[](2);
+        path[0] = WETH;
+        path[1] = DAI;
+        amounts = uniswap.swapExactTokensForTokens(
+            IERC20(WETH).balanceOf(address(this)), 0, path, address(this), now);
+        assertEq(amounts.length, 2);
+        assertEq(amounts[0], ethBal);
+        assertEq(IERC20(WETH).balanceOf(address(this)), 0);
+        assertEq(IERC20(DAI).balanceOf(address(this)), amounts[1]);
+
+        diff = amounts[1] > 10_000 ether ? amounts[1] - 10_000 ether : 10_000 ether - amounts[1];
```

Comment:
I *think* this is different because in the previous check `amounts[1]` is `WETH`, however the route is changed so by the time we get here `amounts[1]` is Dai. I've added comments to reflect this. @lucas-manuel  should confirm.

Reply:
that makes much more sense now